transaction/src/view/action_view/output_view.rs

```diff
@@ -1,6 +1,14 @@
 use penumbra_crypto::{Note, PayloadKey};
 
-pub struct OutputView {
-    pub decrypted_note: Note,
-    pub decrypted_memo_key: PayloadKey,
+use crate::action::Output;
+#[allow(clippy::large_enum_variant)]
+pub enum OutputView {
+    Visible {
+        output: Output,
+        decrypted_note: Note,
+        decrypted_memo_key: PayloadKey,
+    },
+    Opaque {
+        output: Output,
+    },
```

Comment:
These aren't connected to the protos yet, right?

Reply:
They aren't, and I'm not sufficiently familiar with the proto syntax to know how this kind of enum can best be represented in the proto crate - do you have an example?